src/main/java/seedu/storemando/model/item/ExpiryDate.java

```diff
@@ -37,21 +48,42 @@
     public ExpiryDate(String expiryDate) {
         requireNonNull(expiryDate);
         checkArgument(isValidExpiryDate(expiryDate), MESSAGE_CONSTRAINTS);
+        if (expiryDate.equals(NO_EXPIRY_DATE)) {
+            date = null;
+        } else {
+            date = LocalDate.parse(expiryDate);
+        }
         value = expiryDate;
     }
 
+    public LocalDate getDate() {
+        return this.date;
+    }
+
     /**
      * Returns if a given string is a valid expiryDate.
      */
     public static boolean isValidExpiryDate(String test) {
-        return test.matches(VALIDATION_REGEX);
```

Comment:
Is there a reason for to you removing this check?

Reply:
Instead of first checking if the expiry date format is correct using regex, he parses the input right away and catches DateTimeParseException to determine if the specified input is a valid expiry date. I think it will still work as intended.

Reply:
I felt like the VALIDATION_REGEX does not help to check if the date given by the user is a valid date. So, I used LocalDate instead to check if the date is valid, which can be seen in the subsequent lines. Hope this clears things.